frontend/packages/console-dynamic-plugin-sdk/src/extensions/dashboards.ts

```diff
@@ -0,0 +1,469 @@
+    id: string;
+    /** The title will be the name of the tab */
+    title: string;
+  };
+
+  /** Adds a card on a particular dashboard */
+  export type DashboardsCard = {
+    /** The id of the dashboard tab on which the card will be added. */
+    tab: string;
+    /** The grid position of the card on the dashboard. Valid values: “LEFT”/“RIGHT”/ “MAIN”. */
+    position: 'LEFT' | 'RIGHT' | 'MAIN';
+    /** Loader for the corresponding dashboard card component. */
+    loader: EncodedCodeRef;
+    /** Card's vertical span in the column. Ignored for small screens, defaults to 12. */
+    span?: EncodedCodeRef;
+  };
+
+  export type DashboardsCardCodeRefs = {
+    loader: CodeRef<LazyLoader>;
+    span?: CodeRef<DashboardCardSpan>;
+  };
+
+  /** Adds a health subsystem to the status card of Overview dashboard where the source of status is Prometheus */
+  export type DashboardsOverviewHealthPrometheusSubsystem = {
+    /** The display name of the subsystem. */
+    title: string;
+    /** Resolve the subsystem's health */
+    healthHandler: EncodedCodeRef;
+    /** Additional resource which will be fetched and passed to healthHandler */
+    additionalResource?: EncodedCodeRef;
+    /** Loader for popup content. If defined, a health item will be represented as a link which opens popup with given content. */
+    popupComponent?: EncodedCodeRef;
+    /** The title of the popover. */
+    popupTitle?: string;
+    /** Cloud providers which for which the subsystem should be hidden. */
+    disallowedProviders?: string[];
+  };
+
+  export type DashboardsOverviewHealthPrometheusSubsystemCodeRefs = {
+    healthHandler: CodeRef<PrometheusHealthHandler>;
+    additionalResource?: CodeRef<FirehoseResource>;
+    popupComponent?: CodeRef<LazyLoader<PrometheusHealthPopupProps>>;
+  };
+
+  /** Adds a health subsystem to the status card of Overview dashboard where the source of status is a K8s REST API. */
+  export type DashboardsOverviewURLSubsystem = {
+    /** The display name of the subsystem. */
+    title: string;
+    /** The URL to fetch data from. It will be prefixed with base k8s URL */
+    url: string;
+    /** Custom function to fetch data from the URL. If none is specified, default one (coFetchJson) will be used. Response is then parsed by “healthHandler”. */
+    fetch?: string;
+    /** Resolve the subsystem's health */
+    healthHandler: EncodedCodeRef;
+    /** Additional resource which will be fetched and passed to healthHandler */
+    additionalResource?: EncodedCodeRef;
+    /** Loader for popup content. If defined, a health item will be represented as a link which opens popup with given content. */
+    popupComponent?: EncodedCodeRef;
+    /** The title of the popover. */
+    popupTitle?: string;
+  };
+
+  export type DashboardsOverviewURLSubsystemCodeRefs<R = any> = {
+    healthHandler: CodeRef<URLHealthHandler<R>>;
+    additionalResource?: CodeRef<FirehoseResource>;
+    popupComponent?: CodeRef<LazyLoader<any>>;
```

Comment:
we need to add a type here, instead of `any`. The `popupComponent` receives these props https://github.com/openshift/console/blob/master/frontend/public/components/dashboard/dashboards-page/cluster-dashboard/health-item.tsx#L157-L159